airflow/models/variable.py

```diff
@@ -200,8 +200,7 @@ def update(
 
         if Variable.get_variable_from_secrets(key=key) is None:
             raise KeyError(f"Variable {key} does not exist")
-
-        obj = session.query(Variable).filter(Variable.key == key).first()
+        obj = session.scalar(select(Variable).where(Variable.key == key))
```

Comment:
Should we use `limit(1)` since it was picking the first item?

Reply:
`Variable.key` is unique so it shouldn’t matter